diff --git a/Makefile b/Makefile
index c0de461a..c89d4c27 100644
--- a/Makefile
+++ b/Makefile
@@ -3,6 +3,7 @@ include $(shell ocamlfind query visitors)/Makefile.preprocess
 
 .PHONY: all minimal clean test
 
+OCAML_VERSION=$(shell ocaml -vnum)
 OCAMLBUILD=ocamlbuild -I src -I lib -I parser -I krmllib -use-menhir -use-ocamlfind \
  -menhir "menhir --infer --explain" -quiet
 FLAVOR?=native
@@ -57,7 +58,7 @@ test: all
 pre:
 	@which fstar.exe >/dev/null 2>&1 || [ -x $(FSTAR_HOME)/bin/fstar.exe ] || \
 	  { echo "Didn't find fstar.exe in the path or in FSTAR_HOME (which is: $(FSTAR_HOME))"; exit 1; }
-	@ocamlfind query fstarlib >/dev/null 2>&1 || [ -f $(FSTAR_HOME)/bin/fstarlib/fstarlib.cmxa ] || \
+	@ocamlfind query fstar.lib >/dev/null 2>&1 || [ -f $(FSTAR_HOME)/lib/ocaml/$(OCAML_VERSION)/site-lib/fstar/lib/fstar_lib.cmxa ] || \
 	  { echo "Didn't find fstarlib via ocamlfind or in FSTAR_HOME (which is: $(FSTAR_HOME)); run $(MAKE) -C $(FSTAR_HOME)/ulib/ml"; exit 1; }
 
 
diff --git a/_tags b/_tags
index 61c26850..d6231c5e 100644
--- a/_tags
+++ b/_tags
@@ -13,4 +13,4 @@ true: annot, bin_annot, warn(@1-2-3@8..12@14..21@23..29-30@31..38-39-40-41@43@57
 <test>: -traverse
 <krmllib/{extracted,c,out,dist}>: -traverse
 <book/tutorial>: -traverse
-<krmllib/**/*.ml>: package(fstarlib)
+<krmllib/**/*.ml>: package(fstar.lib)
